backend/main/strategies/balance_of_nfts.go

```diff
@@ -0,0 +1,127 @@
+package strategies
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/DapperCollectives/CAST/backend/main/models"
+	"github.com/DapperCollectives/CAST/backend/main/shared"
+	s "github.com/DapperCollectives/CAST/backend/main/shared"
+	"github.com/rs/zerolog/log"
+)
+
+type BalanceOfNfts struct {
+	shared.StrategyStruct
+}
+
+func (b *BalanceOfNfts) FetchBalance(
+	db *s.Database,
+	balance *models.Balance,
+	sc *s.SnapshotClient,
+) (*models.Balance, error) {
+
+	vb := &models.VoteWithBalance{
+		NFTs: []*models.NFT{},
+	}
+
+	var c models.Community
+	if err := c.GetCommunityByProposalId(db, balance.Proposal_id); err != nil {
+		return nil, err
+	}
+
+	nftIds, err := models.GetUserNFTs(db, vb)
+	if err != nil {
+		log.Error().Err(err).Msg("error getting user nfts")
+		return nil, err
+	}
+
+	var contract = &shared.Contract{
+		Name: c.Contract_name,
+		Addr: c.Contract_addr,
+	}
+
+	if len(nftIds) == 0 {
+		nftIds, err := b.FlowAdapter.GetNFTIds(balance.Addr, contract)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, nftId := range nftIds {
+			nft := &models.NFT{
+				ID: nftId.(uint64),
+			}
+			vb.NFTs = append(vb.NFTs, nft)
+		}
+		err = models.CreateUserNFTRecord(db, vb)
```

Comment:
before creating the NFT records, you should make sure that none of the NFT IDs have already been used to vote on this proposal. only create NFT records for the ones that havent been used.